v8-315 does not install

When I run the following command
```bash
brew install homebrew/versions/v8-315  
```

I get this error.

```
==> Installing v8-315 from homebrew/versions
==> Downloading https://github.com/v8/v8-git-mirror/archive/3.15.11.18.tar.gz
Already downloaded: /Library/Caches/Homebrew/v8-315-3.15.11.18.tar.gz
==> make dependencies
Last 15 lines from /Users/arian/Library/Logs/Homebrew/v8-315/01.make:
2016-05-31 12:38:50 -0700

make
dependencies

svn checkout --force http://gyp.googlecode.com/svn/trunk build/gyp \
	    --revision 1501
svn: E170013: Unable to connect to a repository at URL 'http://gyp.googlecode.com/svn/trunk'
svn: E160013: '/svn/trunk' path not found
make: *** [dependencies] Error 1
```

--- a/v8-315.rb
+++ b/v8-315.rb
@@ -1,6 +1,6 @@
 class V8315 < Formula
   desc "Google's open source JavaScript engine"
-  homepage "https://code.google.com/p/v8/"
+  homepage "https://github.com/v8/v8/wiki"
   url "https://github.com/v8/v8-git-mirror/archive/3.15.11.18.tar.gz"
   sha256 "93a4945a550e5718d474113d9769a3c010ba21e3764df8f22932903cd106314d"
 
@@ -14,8 +14,21 @@ class V8315 < Formula
 
   keg_only "Conflicts with V8 in Homebrew/homebrew."
 
+  resource "gyp" do
+    url "https://chromium.googlesource.com/external/gyp.git",
+        :revision => "f7bc250ccc4d619a1cf238db87e5979f89ff36d7"
+  end
+
   def install
-    system "make", "dependencies"
+    (buildpath/"build/gyp").install resource("gyp")
+
+    # fix up libv8.dylib install_name
+    # https://github.com/Homebrew/homebrew/issues/36571
+    # https://code.google.com/p/v8/issues/detail?id=3871
+    inreplace "tools/gyp/v8.gyp",
+              "'OTHER_LDFLAGS': ['-dynamiclib', '-all_load']",
+              "\\0, 'DYLIB_INSTALL_NAME_BASE': '#{opt_lib}'"
+
     system "make", "native",
                    "-j#{ENV.make_jobs}",
                    "library=shared",
@@ -27,5 +40,9 @@ class V8315 < Formula
       lib.install Dir["lib*"]
       bin.install "d8", "lineprocessor", "mksnapshot", "preparser", "process", "shell" => "v8"
     end
+  end
+
+  test do
+    assert_equal "Hello World!", pipe_output("#{bin}/v8 -e 'print(\"Hello World!\")'").chomp
   end
 end

